File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.py
import sys
input = sys.stdin.readline


def DFS(graph, visited, v):
  visited[v] = 1
  print(v, end = ' ')

  for i in range(len(graph[v])):
    if (not visited[graph[v][i]]):
      DFS(graph, visited, graph[v][i])

def BFS(graph, visited, v):
  queue = []

  queue.append(v)

  while (len(queue)):
    p = queue.pop(0)

    if (not visited[p]):
      visited[p] = 1
      print(p, end = ' ')

    for i in range(len(graph[p])):
      if (not visited[graph[p][i]]):
        queue.append(graph[p][i])
    
    
n, m, v = map(int, input().split())

# 그래프 초기화
graph = {}
for i in range(n):
  graph[i + 1] = []

# 간선 정보 입력
for _ in range(m):
  p, q = map(int, input().split())
  graph[p].append(q)
  graph[q].append(p)

# 간선 정보를 정점 크기 순으로 정렬
for i in range(1, n + 1):
  graph[i].sort()

# DFS, BFS 결과 출력
visited = [0] * (n + 1)
DFS(graph, visited, v)
print()

visited = [0] * (n + 1)
BFS(graph, visited, v)
print()

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.js
const fs = require("fs");
const input = fs.readFileSync(0, "utf8").trim().split(/\r?\n/);

const [N, M, V] = input[0].split(" ").map(Number);
const graph = Array.from({ length: N + 1 }, () => []);

for (let i = 1; i <= M; i++) {
  const [a, b] = input[i].split(" ").map(Number);
  graph[a].push(b);
  graph[b].push(a);
}

for (let i = 1; i <= N; i++) graph[i].sort((x, y) => x - y);

const visitedD = Array(N + 1).fill(false);
const dfsOrder = [];
function dfs(u) {
  visitedD[u] = true;
  dfsOrder.push(u);
  for (const v of graph[u]) {
    if (!visitedD[v]) dfs(v);
  }
}

const visitedB = Array(N + 1).fill(false);
const bfsOrder = [];
function bfs(start) {
  const q = [start];
  visitedB[start] = true;
  for (let qi = 0; qi < q.length; qi++) {
    const u = q[qi];
    bfsOrder.push(u);
    for (const v of graph[u]) {
      if (!visitedB[v]) {
        visitedB[v] = true;
        q.push(v);
      }
    }
  }
}

dfs(V);
bfs(V);

console.log(dfsOrder.join(" "));
console.log(bfsOrder.join(" "));